### DISCUSSION TITLE
Access files in WSL filesystem by exe in Windows' filesystem

### DISCUSSION THREAD

**Discussion by Joker-Principal** [microsoft/WSL#7697] [Q&A]
### Version

Microsoft Windows [Version 10.0.19043.1348] 

### WSL Version

- [x] WSL 2
- [ ] WSL 1

### Kernel Version

5.4.72

### Distro Version

Ubuntu 20.04

### Other Software

notepad.exe or any exe located on Windows

### Repro Steps

1. Open terminal and run `wsl.exe`
2. Write a simple C source file: `echo 'int main() { return 0; }' > ~/main.c`
3. run `cd ~ && notepad.exe ~/main.c`, OK
4. run `cd /mnt/c && notepad.exe ~/main.c`, notepad can't find the file

### Expected Behavior

The exe files (in Windows) will work fine when the current working directory is on both Windows and WSL.

### Actual Behavior

Only the working directory is on Linux that exe files can find the files in WSL's filesystem.

**Comment by Biswa96** [comment]
That `~/main.c` path has to be converted to Win style first. i.e. this may work `notepad.exe $(wslpath -w ~/main.c)`

**Comment by Joker-Principal** [comment]
> That `~/main.c` path has to be converted to Win style first. i.e. this may work `notepad.exe $(wslpath -w ~/main.c)`

Yes, it works. But actually, I'm using CMake to compile it. The path sent to compiler is generated by CMake and hard to modify it.

**Comment by Biswa96** [comment]
CMake where?

**Comment by Joker-Principal** [comment]
> CMake where?
WSL /usr/bin

**Comment by Biswa96** [comment]
CMake in Linux environment can access ~/main.c without any path conversion. 

If the CMake in Windows environment you need something like in VSCode or ssh.

**Comment by Joker-Principal** [comment]
> CMake in Linux environment can access ~/main.c without any path conversion.
> 
> If the CMake in Windows environment you need something like in VSCode or ssh.

Yes, I built the project successfully. Thank you!   
Howerver, the problem in issue seems to be not solved... It didn't appear When I used WSL1. So I guess that maybe the exe is executed by Windows directly, in order to optimize the speed of accessing files.
